cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamService.java

```diff
@@ -67,4 +77,23 @@ protected final void shutDown() throws Exception {
     janitorService.stopAndWait();
     streamCoordinatorClient.stopAndWait();
   }
+
+  /**
+   * Create Notification feed for stream's heartbeats, if it does not already exist.
+   */
+  private void createHeartbeatsFeed() throws NotificationFeedException {
+    // TODO worry about namespaces here. Should we create one heartbeat feed per namespace?
```

Comment:
There is a JIRA for implementing namespaces in notifications in general - https://issues.cask.co/browse/CDAP-1030.